std/algorithm/sorting.d

```diff
@@ -3753,57 +3753,45 @@ void topNIndex(alias less = "a < b", SwapStrategy ss = SwapStrategy.unstable,
 if (isRandomAccessRange!Range &&
     isRandomAccessRange!RangeIndex &&
     hasAssignableElements!RangeIndex &&
-    isIntegral!(ElementType!(RangeIndex)))
+        isIntegral!(ElementType!(RangeIndex)) || is(ElementType!(RangeIndex) == ElementType!(Range)*))
```

Comment:
What's with the extra 4 leading spaces?

I suggest adding parens around this line to make the logic grouping clearer.